bin/mail/2017-05-17-end-of-year/generate-list

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env ruby
+
+require_relative '../mailing-common/mailing-list-utils'
+
+PEGASUS_REPORTING_DB_READONLY = sequel_connect(
+  CDO.pegasus_reporting_db_reader,
+  CDO.pegasus_reporting_db_reader
+)
+
+# EMAIL 1 (csf_pd): Code Studio teachers who attended a K5 PD.
+query_1_csf_pd = "
+  SELECT email
+  FROM contact_rollups
+  WHERE opted_out IS NULL
+    AND roles LIKE '%Teacher%'
+    AND professional_learning_attended LIKE '%CS Fundamentals%'
+"
+results_1_csf_pd = {}
+PEGASUS_REPORTING_DB_READONLY.fetch(query_1_csf_pd).
+  reject {|row| UNSUBSCRIBERS.include? row[:email]}.
+  each do |row|
+  results_1_csf_pd[row[:email]] = {email: row[:email]}
+end
+
+puts "LIST 1 (CSF PD) SIZE: #{results_1_csf_pd.size}."
+export_contacts_to_csv results_1_csf_pd, 'email_1_csf_pd.csv'
+
+# EMAIL 2: Code Studio teachers in (US or unknown location) and
+#   (sections where grade is K5 or ages of students taught is 4 to 11)
+# minus EMAIL 1.
+
+query_2_csf_no_pd = "
+  SELECT email
+  FROM contact_rollups
+  WHERE opted_out IS NULL
+    AND roles LIKE '%Teacher%'
+    AND (country = 'United States' OR country IS NULL)
+    AND (
+      (
+        grades_taught LIKE '%K%'
+          OR grades_taught LIKE '%,1,%' OR grades_taught = '1' OR grades_taught LIKE '1,%' OR grades_taught LIKE '%,1'
+          OR grades_taught LIKE '%,2,%' OR grades_taught = '2' OR grades_taught LIKE '2,%' OR grades_taught LIKE '%,2'
+          OR grades_taught LIKE '%,3,%' OR grades_taught = '3' OR grades_taught LIKE '3,%' OR grades_taught LIKE '%,3'
+          OR grades_taught LIKE '%,4,%' OR grades_taught = '4' OR grades_taught LIKE '4,%' OR grades_taught LIKE '%,4'
+          OR grades_taught LIKE '%,5,%' OR grades_taught = '5' OR grades_taught LIKE '5,%' OR grades_taught LIKE '%,5'
```

Comment:
Wow, this is cumbersome. Can we use [FIND_IN_SET](https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_find-in-set) instead?

```
FIND_IN_SET('1', grades_taught) > 0 
OR FIND_IN_SET('2', grades_taught) > 0 
# etc
```